### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [twaugh/patchutils#1]
I have looked at a few source files for your current software. I have noticed that some checks for return codes are missing.

Would you like to add more error handling for return values from functions like the following?
- [fgetpos](http://pubs.opengroup.org/onlinepubs/9699919799/functions/fgetpos.html "Get current file position information") ⇒ [copy_hunk](https://github.com/twaugh/patchutils/blob/44a1a327310eb2a222ae190c183fb146ea8beaac/src/rediff.c#L74 "Copy hunk from in to out with no modifications.")
- [fputs](http://pubs.opengroup.org/onlinepubs/9699919799/functions/fputs.html "Put a string on a stream") ⇒ [output_header_line](https://github.com/twaugh/patchutils/blob/f39134c7b51bf591ff494496cddf9e5cae38c80a/src/filterdiff.c#L155 "Output header line")
- [fwrite](http://pubs.opengroup.org/onlinepubs/9699919799/functions/fprintf.html "Binary output") ⇒ [xopen_seekable](https://github.com/twaugh/patchutils/blob/92255b5934f0eb2465bd9472e900cd460efdd1d8/src/util.c#L199 "Make a file seekable.")

**Comment by twaugh** [comment]
Yes please! If you'd like to do that yourself in a fork, a pull request would be welcomed...

**Comment by elfring** [comment]
I suggest to avoid [ignorance of return values](https://cwe.mitre.org/data/definitions/252.html "Unchecked return values") a bit more.

Are you interested to apply [aspect-oriented software development](http://aspectc.org/ "AspectC++")?
How do you think about to encapsulate error detection and corresponding exception handling as [a reusable aspect](http://en.wikipedia.org/wiki/Aspect_%28computer_programming%29 "Combination of a pointcut with advice") in your software?
